kip-0017.md

```diff
@@ -8,105 +8,425 @@ Category: Interface
+- `kadena:mainnet01` - Kadena mainnet
+- `kadena:testnet04` - Current kadena testnet
+- `kadena:development` - Kadena devnet
+
+> **NB**: `Chain IDs`, as defined here, are semantically specific to
+> WalletConnect and are not equivalent to Kadena's "chain IDs".
+> They are more closely related to Kadena's `networkId` field.
+> <Read the motivation for this in TODO>
 
-Here are some example accounts:
 
+## Wallet Connect Accounts
+
+Accounts are Wallet Connect's way of specifying control over signing priviledges.
+
+They are defined by the [CAIP-10 standard](https://github.com/ChainAgnostic/CAIPs/blob/master/CAIPs/caip-10.md), and are formatted as:
+```
+    <chain_id> + ":" + <account_address>
 ```
+- `<chain_id>`: `string` - the Wallet Connect [Chain ID](#wallet-connect-chain-ids).
+- `<account_address>`: `string` - the public key available for signing by the wallet.
+
+> **NB**: `accounts` is semantically specific to WalletConnect. This does not
+> represent an account in a Kadena fungible contract (e.g. `coin`).
+> Read the motivation for this in [Accounts vs. Public Keys](#accounts-vs-public-keys).
+
+Some users will want to know which account names in a Kadena fungible contract are associated with the public keys returned by Wallet Connect.
+
+These users have two options:
+- Some account names can be derived from the public key (e.g. `k:` accounts that are defined as `k:<public_key>`).
+- For all other cases, the [`kadena_getAccounts_v1`](#kadena_getAccounts_v1)
+method can be called.
+
+## WalletConnect Methods
+
+Wallet Connect methods are ways for dApps to interact with a wallet using predefined method identifiers and an optional payload.
+
+The methods available for Kadena are:
+
+- [`kadena_getAccounts_v1`](#kadena_getAccounts_v1)
+- [`kadena_quickSign_v1`](#kadena_quicksign_v1)
+- [`kadena_sign_v1`](#kadena_sign_v1)
+
+### kadena_getAccounts_v1
+
+This method returns the Kadena, fungible account names associated with a specific Wallet Connect account (a prefixed public key). 
 
-kadena:mainnet01:22ddc64851718e9d41d98b0f33d5e328ae5bbbbd97aed9885adac0f2d070ff9c
+These Wallet Connect accounts are retrieved from the `accounts` property of a [Paring response](#pairing-with-walletconnect).
 
-kadena:testnet04:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89
+#### Example kadena_getAccounts_v1 Request and Response
+
+```jsonc
+// kadena_getAccounts_v1 Method Request
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "kadena_getAccounts_v1",
+  "params": {
+    "accounts": [
+      {
+        "account": "kadena:mainnet01:38298612cc2d5e841a232bd08413aa5304f9ef3251575ee182345abc3807dd89",
+        "contracts": ["coin"] // optional, when omitted, the wallet returns all known fungible accounts
+      }
+    ]
+  }
+}
+````
+
+```jsonc
+// kadena_getAccounts_v1 Method Response
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": [
```

Comment:
Added the "id", "jsonrpc" fields to getAccounts request and response examples because I couldn't find the Wallet Connect Data Structure used by this Cosmos example when describing their methods: https://docs.walletconnect.com/2.0/advanced/rpc-reference/cosmos-rpc#example